From 428f86ee9190c5386e8a86b80a05fb67935037ee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Wed, 19 Feb 2025 06:25:06 +0100
Subject: [PATCH] fix(tests): Filter out lines with __typeof__ macro

Problem: musl libc makes heavy use of __typeof__ as part of its redir
macro for backwards compatibility with 32-bit time_t values.
Unfortunately, the __typeof__ keyword is not supported by LuaJIT C
parser.

Solution: Filter out the keyword in filter_complex_blocks.
---
 test/unit/testutil.lua | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/unit/testutil.lua b/test/unit/testutil.lua
index 7fe27ad735..1f635f277b 100644
--- a/test/unit/testutil.lua
+++ b/test/unit/testutil.lua
@@ -146,6 +146,8 @@ local function filter_complex_blocks(body)
         or string.find(line, 'value_init_')
         or string.find(line, 'UUID_NULL') -- static const uuid_t UUID_NULL = {...}
         or string.find(line, 'inline _Bool')
+        -- used by musl libc headers via __REDIR marco
+        or string.find(line, '__typeof__')
         -- used by macOS headers
         or string.find(line, 'typedef enum : ')
         or string.find(line, 'mach_vm_range_recipe')
